packages/@aws-cdk/aws-apigateway/lib/stepfunctions-api.ts

```diff
@@ -0,0 +1,154 @@
+   *      stateMachine: stateMachine,
+   *      requestContext: {
+   *         accountId: true,
+   *         apiId: true,
+   *         apiKey: true,
+   *         authorizerPrincipalId: true,
+   *         caller: true,
+   *         cognitoAuthenticationProvider: true,
+   *         cognitoAuthenticationType: true,
+   *         cognitoIdentityId: true,
+   *         cognitoIdentityPoolId: true,
+   *         httpMethod: true,
+   *         stage: true,
+   *         sourceIp: true,
+   *         user: true,
+   *         userAgent: true,
+   *         userArn: true,
+   *         requestId: true,
+   *         resourceId: true,
+   *         resourcePath: true,
+   *       },
+   *    });
+   *
+   * @default - all parameters within request context will be set as false
+   */
+  readonly requestContext?: RequestContext;
+}
+
+/**
+ * Defines an API Gateway REST API with a Synchrounous Express State Machine as a proxy integration.
+ */
+export class StepFunctionsRestApi extends RestApi {
+  constructor(scope: Construct, id: string, props: StepFunctionsRestApiProps) {
+    if (props.defaultIntegration) {
+      throw new Error('Cannot specify "defaultIntegration" since Step Functions integration is automatically defined');
+    }
+
+    if ((props.stateMachine.node.defaultChild as sfn.CfnStateMachine).stateMachineType !== sfn.StateMachineType.EXPRESS) {
+      throw new Error('State Machine must be of type "EXPRESS". Please use StateMachineType.EXPRESS as the stateMachineType');
+    }
+
+    const apiRole = role(scope, props);
+    const methodResp = methodResponse();
+
+    let corsEnabled;
+
+    if (props.defaultCorsPreflightOptions !== undefined) {
+      corsEnabled = true;
+    } else {
+      corsEnabled = false;
+    }
+
+    super(scope, id, {
+      defaultIntegration: new StepFunctionsIntegration(props.stateMachine, {
+        credentialsRole: apiRole,
+        corsEnabled: corsEnabled,
+        requestContext: props.requestContext,
+        action: 'StartSyncExecution',
+      }),
+      ...props,
+    });
+
+    if (!corsEnabled) {
+      this.root.addMethod('ANY', new StepFunctionsIntegration(props.stateMachine, {
+        credentialsRole: apiRole,
+        requestContext: props.requestContext,
+        action: 'StartSyncExecution',
+      }), {
+        methodResponses: [
+          ...methodResp,
+        ],
+      });
+    }
+  }
+}
+
+function role(scope: Construct, props: StepFunctionsRestApiProps): iam.Role {
+  const apiName: string = props.stateMachine + '-apiRole';
+  const apiRole = new iam.Role(scope, apiName, {
+    assumedBy: new iam.ServicePrincipal('apigateway.amazonaws.com'),
+  });
+
+  apiRole.attachInlinePolicy(
+    new iam.Policy(scope, 'AllowStartSyncExecution', {
+      statements: [
+        new iam.PolicyStatement({
+          actions: ['states:StartSyncExecution'],
+          effect: iam.Effect.ALLOW,
+          resources: [props.stateMachine.stateMachineArn],
+        }),
+      ],
+    }),
+  );
```

Comment:
We should be using `grant` APIs instead, such as [`grantExecution()`](https://github.com/aws/aws-cdk/blob/5bf0d074854ff90c5d9521f5d7f0fc9ff31c5eb5/packages/%40aws-cdk/aws-stepfunctions/lib/state-machine.ts#L218).

Looks like `grantExecution()` needs to be updated to provide `StartSyncExecution`.